activator/src/activator.rs

```diff
@@ -116,18 +106,55 @@ pub async fn get_snapshot_poll(
     Ok(())
 }
 
-pub fn process_events_thread(
+pub async fn websocket_task(
     client: Arc<DZClient>,
     tx: mpsc::Sender<(Box<Pubkey>, Box<AccountData>)>,
     stop_signal: Arc<AtomicBool>,
-) -> eyre::Result<()> {
-    client.subscribe(
-        |_, pubkey, data| {
-            tx.blocking_send((pubkey, data)).unwrap_or_else(|err| {
-                log::error!("Failed to send websocket data to processor: {}", err);
-            });
-        },
-        stop_signal,
-    )?;
-    Ok(())
+) {
+    while !stop_signal.load(std::sync::atomic::Ordering::Relaxed) {
+        info!(
+            "Starting websocket task on {}, program_id: {}",
+            client.get_ws(),
+            client.get_program_id()
+        );
+        match AsyncDZClient::new(client.get_ws(), *client.get_program_id()).await {
+            Ok(async_client) => match async_client.subscribe().await {
+                Ok((mut subscription, unsubscribe)) => {
+                    info!("Websocket subscription established");
+                    while !stop_signal.load(std::sync::atomic::Ordering::Relaxed) {
+                        if let Some(msg) = subscription.next().await {
+                            let keyed_account = msg.value;
+                            let pubkey = keyed_account.pubkey.clone();
+                            match rpckeyedaccount_decode(keyed_account) {
+                                Ok(Some((pubkey, account))) => {
+                                    tx.send((pubkey, account)).await.unwrap();
```

Comment:
You could do something like:

```rust
if let Err(e) = tx.send(...).await {
    print_some_error_or_whatever
    break;
}
```

The unwrap would crash the entire activator process otherwise